crates/proof-of-sql/src/proof_primitive/dory/dynamic_dory_commitment_helper_gpu.rs

```diff
@@ -27,13 +45,18 @@ use tracing::{span, Level};
 #[tracing::instrument(
     name = "compute_dynamic_dory_commitments (gpu)",
     level = "debug",
-    skip_all
+    skip_all,
+    fields(total_memory, used_memory)
```

Comment:
Percent used was a nice metric to have.
Can we have `start_` and `end_` versions of these as well?
In other words, put a `log_memory_start_usage()` and `log_memory_end_usage()`.

That way we can see the changed that occurred within a single span just by looking at that span.

Reply:
Updated, but I'm not sure if it fits with what you were thinking?

Reply:
Does this work? Does it show all 6 things in the span inside jaeger?

I was expecting `fields(available_memory_start, used_memory_start, percentage_memory_used_start, available_memory_end, used_memory_end, percentage_memory_used_end)` or something. But, if this works, I like this better.

Reply:
Here is what it looks like with just `fields(available_memory, used_memory, percentage_memory_used)`, from [this commit](https://github.com/spaceandtimelabs/sxt-proof-of-sql/pull/412/commits/5a5c190376f228d2866f7bc87b9e4450eb9c2d34). ![image](https://github.com/user-attachments/assets/857a542b-2efe-4dd6-a64e-98b1e9422eed)